doc/introduction/architecture.md

```diff
@@ -30,29 +34,40 @@ The OCR JSON is used to extract many types of insights:
 - brand
 - ...
 
-We use simple string matching algorithms to find patterns in the OCR text to generate new insights [^insights]. These insights are stored in the PostgreSQL database.
+Predictions, as well as insights are stored in the PostgreSQL database.
+
+[^predictions]: see `robotoff.models.Prediction`
+
+[^insights]: see `robotoff.models.ProductInsight`
 
-[^insights]: see `models.ProductInsight`
+[^logos]: see `robotoff.logos`
+
+These new insights are then accessible to all annotation tools (Hunger Games, mobile apps,...), that can validate or not the insight. 
+
+If the insight is validated by an authenticated user, it's applied immediately and the product is updated through Product Opener API [^annotate]. If it's reported as invalid, no update is performed, but the insight is marked as annotated, so that it is not suggested to another annotater. If the user is not authenticated, a system of votes is used (3 consistent votes).
```

Comment:
```suggestion
If the insight is validated by an authenticated user, it's applied immediately and the product is updated through Product Opener API [^annotate]. If it's reported as invalid, no update is performed, but the insight is marked as annotated so that it is not suggested to another annotator. If the user is not authenticated, a system of votes is used (3 consistent votes trigger the insight application).
```